arborist/server.go

```diff
@@ -468,13 +488,29 @@ func (server *Server) handleAuthRequest(w http.ResponseWriter, r *http.Request,
 func (server *Server) handleListAuthResourcesGET(w http.ResponseWriter, r *http.Request) {
 	authRequest := &AuthRequest{}
 	var errResponse *ErrorResponse
-	authRequest, errResponse = authRequestFromGET(server.decodeToken, r)
-	if errResponse != nil {
-		errResponse.log.write(server.logger)
-		_ = errResponse.write(w, r)
+	hasJWT := r.Header.Get("Authorization") != ""
+	usernameInJWT := false
+	if hasJWT {
+		authRequest, errResponse = authRequestFromGET(server.decodeToken, r)
+		usernameInJWT = authRequest.Username != ""
+		if errResponse != nil {
```

Comment:
do `if errResponse != nil {` before `usernameInJWT = authRequest.Username != ""` to avoid possible crash